cpp/daal/src/algorithms/cordistance/cordistance_kernel.h

```diff
@@ -44,6 +44,9 @@ class DistanceKernel : public Kernel
 public:
     services::Status compute(const size_t na, const NumericTable * const * a, const size_t nr, NumericTable * r[],
                              const daal::algorithms::Parameter * par);
+
+    services::Status compute(const size_t na, const NumericTable * const a, const size_t nb, const NumericTable * const b, const size_t nr,
+                             NumericTable * r, const daal::algorithms::Parameter * par);
```

Comment:
Sorry, I didn't notice it earlier. But this function is not needed at all.
The original semantics of
`compute(const size_t na, const NumericTable * const * a, const size_t nr, NumericTable * r[],  const daal::algorithms::Parameter * par)`
means:
- `na` - number of input data tables;
- `a` - array of input data tables;
- `nr` - number of data tables with results;
- `r` -array of resulting data tables.

So, you can pass both `x` and `y` data tables using `na == 2` and `const NumericTable a[2] = [daal_x.get(), daal_y.get()];`.

Reply:
I originally implemented this exact suggestion, leveraging the original function prototype from [cpp/daal/src/algorithms/cordistance/cordistance_kernel.h](https://github.com/uxlfoundation/oneDAL/pull/3131/files/09f668b002d685eed07596c864b6857aae13da46#diff-15d3f0cf6ecaa55c27ea3f0f113480f84c33419e8bb25e509ad7cecde2b41af8) in [this commit](https://github.com/uxlfoundation/oneDAL/pull/3131/commits/de59d01412dfd1b8b6f73620114ad86f4c9298fb). However, I changed it based on your previous comments, suggesting I should preserve the original function and create a new one. Has your stance changed or was there a misunderstanding from the previous code changes I made?

Reply:
There was a misunderstanding. You have to preserve the original functions `corDistanceFull`, `corDistanceLowerPacked` and `corDistanceUpperPacked` indeed.
But you have to create 1 new `corDistanceFull` function that accepts both X and Y tables. There is no need in creating new `compute` function for it, you can just do following changes in the existing compute:

```
if (isFull<algorithmFPType, cpu>(rLayout))
{
    if (na == 1)
    {
        return corDistanceFull<algorithmFPType, cpu>(xTable, rTable);
    } 
    else if (na == 2)
    {
        NumericTable * yTable = const_cast<NumericTable *>(a[1]);
        return corDistanceFull<algorithmFPType, cpu>(xTable, yTable, rTable);
    }
}
```